example-repo/packages/remix-basic/vite.config.ts

```diff
@@ -0,0 +1,29 @@
+import { dmnoRemixVitePlugin, dmnoRemixPreset } from "@dmno/remix-integration";
+import { vitePlugin as remix } from "@remix-run/dev";
+import { defineConfig } from "vite";
+import tsconfigPaths from "vite-tsconfig-paths";
+
+// console.log('> secret value =', DMNO_CONFIG.SECRET_STATIC);
+
+console.log('loaded vite.config.ts');
+
+export default defineConfig(({ command, mode, isSsrBuild }) => {
+  console.log('VITE DEFINE CONFIG!', { command, mode, isSsrBuild });
+  
+  // defineConfigArgs.isSsrBuild - will be useful
+
+  return {
+    plugins: [
+      dmnoRemixVitePlugin({ command, mode, isSsrBuild }) as any,
+      remix({
+        future: {
+          v3_fetcherPersist: true,
+          v3_relativeSplatPath: true,
+          v3_throwAbortReason: true,
+        },
+        presets: [dmnoRemixPreset() as any],
```

Comment:
one slight bummer, currently we have to add the preset and the vite plugin, although this is only necessary to inject the public+dynamic config api route. Alternatively we could have people just add the route file more explicitly.